src/js/providers/html5.js

```diff
@@ -998,6 +1015,48 @@ define([
                 _this.trigger('mediaType', {mediaType: mediaType});
             }
         }
+
+        // If we're live and the buffer end has remained the same for some time, mark the stream as stale and check if the stream is over
+        function checkStaleStream() {
+            var endOfBuffer = timeRangesUtil.endOfRange(_videotag.buffered);
+            var live = (_videotag.duration === Infinity);
+
+            if (live && _lastEndOfBuffer === endOfBuffer) {
+                if (!_staleStreamTimeout) {
+                    _staleStreamTimeout = setTimeout(function () {
+                        _stale = true;
+                        checkStreamEnded();
+                    }, _staleStreamDuration);
+                }
+            } else {
+                clearTimeout(_staleStreamTimeout);
```

Comment:
This should probably be called anywhere we call `clearTimeout(_playbackTimeout);`